axelor-account/src/main/java/com/axelor/apps/account/service/AccountingReportValueServiceImpl.java

```diff
@@ -0,0 +1,848 @@
+import com.axelor.apps.account.db.repo.AccountingReportTypeRepository;
+import com.axelor.apps.account.db.repo.AccountingReportValueRepository;
+import com.axelor.apps.account.db.repo.MoveLineRepository;
+import com.axelor.apps.account.db.repo.MoveRepository;
+import com.axelor.apps.account.exception.AccountExceptionMessage;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.common.StringUtils;
+import com.axelor.db.Model;
+import com.axelor.db.Query;
+import com.axelor.exception.AxelorException;
+import com.axelor.exception.db.repo.TraceBackRepository;
+import com.axelor.rpc.Context;
+import com.axelor.script.GroovyScriptHelper;
+import com.axelor.script.ScriptHelper;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.commons.collections.CollectionUtils;
+
+public class AccountingReportValueServiceImpl implements AccountingReportValueService {
+  protected AppBaseService appBaseService;
+  protected AccountingReportValueRepository accountingReportValueRepo;
+  protected MoveLineRepository moveLineRepo;
+
+  protected int lineOffset = 0;
+
+  @Inject
+  public AccountingReportValueServiceImpl(
+      AppBaseService appBaseService,
+      AccountingReportValueRepository accountingReportValueRepo,
+      MoveLineRepository moveLineRepo) {
+    this.appBaseService = appBaseService;
+    this.accountingReportValueRepo = accountingReportValueRepo;
+    this.moveLineRepo = moveLineRepo;
+  }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void clearReportValues(AccountingReport accountingReport) {
+    accountingReportValueRepo.findByAccountingReport(accountingReport).remove();
+  }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void computeReportValues(AccountingReport accountingReport) throws AxelorException {
+    Map<String, Map<String, AccountingReportValue>> valuesMapByColumn = new HashMap<>();
+    Map<String, Map<String, AccountingReportValue>> valuesMapByLine = new HashMap<>();
+
+    AccountingReportType accountingReportType = accountingReport.getReportType();
+    this.checkAccountingReportType(accountingReportType);
+
+    boolean isAllComputed = false;
+    int timeoutCounter = 0;
+
+    while (!isAllComputed) {
+      isAllComputed = this.createReportValues(accountingReport, valuesMapByColumn, valuesMapByLine);
+
+      if (timeoutCounter++ > appBaseService.getProcessTimeout()) {
```

Comment:
ProcessTimeout is in seconds normally, not a number of iterations. Here per example, if ProcessTimeout = 10s, after 10 iterations, we'll get the exception, but I'm sure it would have taken a lot less than 10s to do 10 iterations.